#VRML_SIM R2021b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.09310924936350908 0.9715714608922188 0.21766847282673588 2.3538345567544585
  position 0.4647562343030063 0.5181907029577503 -0.4934229756833013
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 3 3
}
Solid {
  translation -0.8 0.48767239042709126 0.8
  rotation 0.6629355666793558 0.7434996248873883 -0.08789051270663041 8.469092921943637e-17
  children [
    DEF cylinder Shape {
      appearance PBRAppearance {
        baseColor 1 1 0.8
        roughness 0.8
        metalness 0.5
      }
      geometry Cylinder {
        height 1
        radius 0.1
      }
    }
  ]
  name "column1"
  boundingObject USE cylinder
  physics Physics {
  }
}
Solid {
  translation -0.8 0.25 -0.8
  children [
    USE cylinder
  ]
  name "column4"
}
Solid {
  translation 0.8 0.25 -0.8
  children [
    USE cylinder
  ]
  name "column3"
}
Solid {
  translation 0.8 0.25 0.8
  children [
    USE cylinder
  ]
  name "column2"
}
Solid {
  translation 0.1 0.2 0.5
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.7 0.7
        metalness 0.2
      }
      geometry Box {
        size 1 0.4 0.05
      }
    }
  ]
  name "wall1"
}
Solid {
  translation 0.57 0.2 -5.81174e-07
  rotation 0 1 0 -1.5707953071795862
  children [
    USE wall
  ]
  name "wall2"
}
Robot {
  translation 0 0.22 0
  rotation 0 1 0 1.57
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        position 1.5697769769088893e-06
        axis 0 0 1
        anchor 0.08 -0.18 -0.06
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.08000000000014795 -0.18000009418639712 -0.0599997174400703
        rotation 0.9999999999993836 7.84888488454122e-07 7.855154219375929e-07 1.5699978908695498
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0.7 0.9
              roughness 0.6
            }
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        name "front_left"
        boundingObject USE wheel
        physics DEF physics_wheel Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2.08511922558314e-06
        anchor 0.08 -0.18 0.06
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation 0.08 -0.17999987489323777 0.060000375321591054
        rotation 1 0 0 1.570002737436035
        children [
          USE wheel
        ]
        name "front_right"
        boundingObject USE wheel
        physics USE physics_wheel
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -1.0411010647052084e-06
        anchor -0.08 -0.18 -0.06
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation -0.08 -0.17999993753403376 -0.06000018739822421
        rotation 0.9999999999999999 0 0 1.5699990611623285
        children [
          USE wheel
        ]
        boundingObject USE wheel
        physics USE physics_wheel
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -1.0008787096005278e-05
        anchor -0.08 -0.18 0.06
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.08 -0.1800006005362416 0.059998198421327996
        rotation 0.9999999999999999 0 0 1.569996141172255
        children [
          USE wheel
        ]
        name "back_right"
        boundingObject USE wheel
        physics USE physics_wheel
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
        baseColor 1 0.9 1
        baseColorMap ImageTexture {
          url [
            "F:/code file/robotics/suitcase.jpg"
          ]
          filtering 1
        }
        roughness 0.3
        metalness 0.4
      }
      geometry Box {
        size 0.2 0.35 0.1
      }
    }
    DistanceSensor {
      translation 0.1 0 -0.03
      rotation 0 -1 0 -0.3
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0.7 0.6 0.9
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.1 0 0.03
      rotation 0 1 0 -0.2235953071795862
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0.7 0.6 0.9
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor
      physics Physics {
      }
    }
  ]
  name "suitcase"
  boundingObject USE body
  physics Physics {
  }
  controller "obstacle_avoidance"
}
